components/fxa-client/src/fxa_client.udl

```diff
@@ -101,7 +108,7 @@ interface FirefoxAccount {
   //
   // This method constructs as new [`FirefoxAccount`] instance configured to connect
   // the application to a user's account.
-  constructor(FxaConfig config);
+  constructor(FxaConfig config, OAuthHandler oauth_handler);
```

Comment:
Do you think it's worth it for us to also comment on the limitation of callback interfaces when it comes to releasing memory?

Particularly whatever callback interface is passed here will be held on to and **must not be destroyed** while the object is alive?

This makes me wonder how the JVM handles that 🤔

Reply:
I'm pretty sure UniFFI ensures that the callback interface stays alive as long as Rust has a reference to it.  For example [CallbackInterfaceRuntime.kt](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/bindings/kotlin/templates/CallbackInterfaceRuntime.kt) keeps an entry in a map to keep it alive.   At some point I want to refactor it to use [UniFfiHandleMap](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/bindings/kotlin/templates/Helpers.kt#L130-L154) which does something similar.

Reply:
I think we want to refactor it to use traits (and deprecate callbacks while we are at it), then it's just an `Arc<>` like everything else :)